flags/src/main/java/com/github/secondbase/flags/Flags.java

```diff
@@ -432,10 +433,13 @@ public Flags parse(final String[] args) {
                     throw new RuntimeException("Could not parse property-file", e);
                 }
             }
+
             Collections.addAll(newArgs, args);
-            optionSet = optionParser.parse(newArgs.toArray(new String[newArgs.size()]));
+            allArgs = newArgs.toArray(new String[newArgs.size()]);
         }
 
+        optionSet = optionParser.parse(fetchSecrets(allArgs));
```

Comment:
I found this a bit difficult to read. But I think I get it now. We could also initialise allArgs as final and assign it in the else block:
{
... 
} else {
   allArgs = args;
}

But I'm not sure this would be better really.